src/pudl/analysis/allocate_net_gen.py

```diff
@@ -21,46 +21,83 @@
+There are six main stages of the allocation process in this module:
+
+* Inputs! Grabbing the input tables from :class:`pudl.output.pudl.PudlTabl`. Some
+  cleaning (like :func:`distribute_annually_reported_data_to_months_if_annual`) and
+  aggregation (like :func:`allocate_bf_data_to_gens`) happen in this stage.
+* Associating the data from the input tables with generator/prime mover/energy source
+  codes - using the primary keys in :py:const:`IDX_GENS_PM_ESC` (see
+  :func:`associate_generator_tables`).
+* Prepare for the allocation (see :func:`prep_alloction_fraction`) by adding boolean
+  flags for whether a particular :py:const:`IDX_GENS_PM_ESC` shows up in each of the
+  input data tables.
+* Allocation! Do the actual allocating. See :func:`allocate_net_gen_by_gen_esc` for net
+  generation allocation and :func:`allocate_fuel_by_gen_esc` for fuel allocation. More
+  details on the allocation process below.
+* Tests! The high-level test of whether the allocated data matches the original inputs
+  is :func:`test_the_original_gf_vs_the_allocated_by_gens_gf`. There are also other
+  warnings that are logged if assumptions aren't met via :func:`_test_frac`,
+  :func:`_test_gen_fuel_allocation` and :func:`_test_gen_pm_fuel_output`
+* Aggregate the main output with primary keys of :py:const:`IDX_GENS_PM_ESC` to the
+  generator-level with primary keys of :py:const:`IDX_GENS` via
+  :func:`aggregate_gen_fuel_by_generator`.
+
+MOAR High-level description about the allocaiton step:
+
+We allocate the net generation/fuel consumption reported in the
+generation_fuel_eia923/boiler_fuel_eia923 table on the basis of plant, prime mover,
+and fuel type among the generators in each plant that have matching fuel types.
+Generation/fuel consumption is allocated proportional to reported generation/fuel
+consumption if it's available, and proportional to each generator's capacity if
+generation/fuel consumption is not available.
+
+In more detail, within each reporting period, we split the plants into three groups:
+
+* Plants where ALL generators report in the more granular generation_eia923 /
+  boiler_fuel_eia923 table.
+* Plants where NONE of the generators report in the generation_eia923 /
+  boiler_fuel_eia923 table.
+* Plants where only SOME of the generators report in the generation_eia923 /
+  boiler_fuel_eia923 table.
+
+In plant-report_date's where ALL generators report more granular generation or fuel
+consumption, the total net generation or fuel consumption reported in the
+generation_fuel_eia923 table is allocated in proportion to the generation each generator
+reported in the generation_eia923 table or fuel consumption reported in the
+boiler_fuel_eia923 table. We do this instead of using net_generation_mwh from
+generation_eia923 or using fuel_consumed_mmbtu from boiler_fuel_eia923 because there are
+some small discrepancies between the total amounts of generation/fuel reported in these
```

Comment:
Discrepancies between which tables? Here it sounds like you're saying there are discrepancies between the `generation_eia923` and `boiler_fuel_eia923` tables, but those aren't really comparable.  I think what you mean is there are discrepancies between the generation reported in the `generation_eia923` and the `generation_fuel_eia923` tables, and similarly discrepancies between the `boiler_fuel_eia923` and `generation_fuel_eia923` tables.

Since the structures & processes associated with the generation and fuel consumption data are so similar, and you need to refer to them both together many times in the documentation, and actually saying both of them every time is unwieldy and hard to read, it might make sense to give them a generic collective name early on, and use that whenever you're talking about both, and only use the individual `generation_eia923` and `boiler_fuel_eia923` names when you're being specific to one or the other of them. Maybe something like "the granular tables?"